src/lib/passwordGenerator.ts

```diff
@@ -0,0 +1,67 @@
+interface passwordProps {
+  includeUpper : boolean;
+  includeLower : boolean;
+  includeSymbol : boolean;
+  includeNumber : boolean;
+}
+
+export default class PasswordGenerator {
+  characters : string = "";
+  passwordLen : number = 12;
+
+  constructor(passwordProps : passwordProps, pwdLength ?: number) {
+    const { includeUpper, includeLower, includeSymbol, includeNumber } =
+      passwordProps;
+    if (pwdLength) this.passwordLength = pwdLength;
+    this.upperCase = includeUpper;
+    this.lowerCase = includeLower;
+    this.symbols = includeSymbol;
+    this.number = includeNumber;
+  }
+
+  set upperCase(isUpperCase : boolean) {
+    if (isUpperCase) {
+      this.characters += "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    }
+  }
+
+  set lowerCase(isLowerCase : boolean) {
+    if (isLowerCase) {
+      this.characters += "abcdefghijklmnopqrstuvwxyz";
+    }
+  }
+
+  set symbols(isSymbol : boolean) {
+    if (isSymbol) {
+      this.characters += "!@#$%^&*";
+    }
+  }
+
+  set number(isNumeric : boolean) {
+    if (isNumeric) {
+      this.characters += "0123456789";
+    }
+  }
+
+  set passwordLength(length : number) {
+    this.passwordLen = length;
+  }
+
+  private _getRandomInteger(min : number, max : number) : number {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
```

Comment:
I think for the purposes of this project using `Math.random()` is probably fine, but using the [pseudo-random number generator from Crypto.subtle](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues) might be more secure.